### DISCUSSION TITLE
TM1914

### DISCUSSION THREAD

**Discussion by bigdingoTECH** [Makuna/NeoPixelBus#778] [Q&A] [answered]
Hi all, and Makuna in particular (you legend).

I understand reading back through various issues threads that support for TM1914 has been added, I would like to test this with some ICs I have here with an ESP32, what is the best example to use for this and how do I configure it.

Best,

M

**Comment by Makuna** [comment, marked as answer]
Just use the NeoPixelTest example, modify it just a little.

Use the specific TM1914 features `NeoRgbTm1914Feature` or `NeoGrbTm1914Feature`.  If you need other color orders just create an issue, they are simple to add.

Use the specific TM1914 method `NeoTm1914Method` or if you have an external inverting circuit (often for level adjustment) then `NeoTm1914InvertedMethod`.  Note that all the normal ESP32 specific variants are available like `NeoEsp32I2s0Tm1914Method` and `NeoEsp32RmtNTm1914Method` and even the ESp32 parallel modes like `NeoEsp32I2s1X8Tm1914Method`.

And make sure to call the `strip.SetPixelSettings(NeoTm1914Settings())` once before your first call to `Show()`

**Comment by bigdingoTECH** [comment]
Thank you :)
